### PR TITLE
auth routes/geocode/twinword

### PR DESCRIPTION
**Pull request by benzeg**
1. Implemented authentication routes
2. Implemented and tested geocode translation to long/lat
3. Implemented and tested twinword word association api

### CODE REVIEW

File: server/lib/api_config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+exports.wordAssocAPIk =
```
Comment (sscchan) [inline server/lib/api_config.js]: You'll need to add the api_config.js with the API keys to .gitignore. Don't want to upload API keys to public repo (we'll probably need to regen these keys)

File: package_REMOTE_6520.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (sscchan) [inline package_REMOTE_6520.json]: Can you clarify the purpose of this particular .json file?

File: package_LOCAL_6520.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
```
Comment (sscchan) [inline package_LOCAL_6520.json]: Can you clarify the purpose of this particular .json file?

File: package_BASE_6520.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (sscchan) [inline package_BASE_6520.json]: Can you clarify the purpose of this particular .json file?

File: package_BACKUP_6520.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  "bugs": {
+    "url": "https://github.com/HR-Proudfoots/proudfoots/issues"
+  },
+  "homepage": "https://github.com/HR-Proudfoots/proudfoots#readme",
+  "dependencies": {
+    "babel-core": "^6.22.1",
+    "babel-loader": "^6.2.10",
+    "babel-preset-es2015": "^6.22.0",
+    "babel-preset-react": "^6.22.0",
+<<<<<<< HEAD
```
Comment (sscchan) [inline package_BACKUP_6520.json]: You'll need to deal with the rebase clash code here prior to git add...

File: server/lib/request-handler.js [hunk 6]
```diff
@@ -1,9 +1,70 @@
...
 
+/////////////////////////////////////////////////////////////////
+//AUTHENTICATION
 
+exports.userLogin = function(req, res) {
+  var username = req.body.username;
+  var password = req.body.password;
+
+  User.findOne( {username: username} )
+    .exec(function(err, user) {
```
Comment (sscchan) [inline server/lib/request-handler.js:16]: Need to handle (err) from the db query (eg. error accessing DB), (and not just when username has length === 0).

File: server/lib/request-handler.js [hunk 7]
```diff
@@ -1,9 +1,70 @@
...
+    ', ',
+    userAddress['city'],
+    ', ',
+    userAddress['state']
+    );
+
+  util.geoCodeit(res, streetAddress, function(err, response) {
+    if (!err) {
+      var geoLocation = response.json.results[0].geometry.location;
+      res.send(geoLocation);
```
Comment (sscchan) [inline server/lib/request-handler.js]: is the geoLocation object identical to the one we discussed eg {location: {lat: <lat>, long: <long>}}, ? Google Map has a convention writing the longitude as the 'lng' property.

File: server/lib/request-handler.js [hunk 8]
```diff
@@ -13,17 +74,17 @@ exports.termSearch = function(req, res) {
...
-    .header("X-Mashape-Key", wordAssocAPIk['key'])
+    .header("X-Mashape-Key", apiKey.wordAssocAPIk['key'])
     .header("Content-Type", "application/x-www-form-urlencoded")
     .header("Accept", "application/json")
     .send(searchTerm)
     .end(function (result) {
       //sends word association back to client as an array of words
-      res.writeHead(200).send(result.body['associations_array']);
+      res.send(result.body['associations_array']);
   });
```
Comment (sscchan) [inline server/lib/request-handler.js]: Need to handle error conditions from Twinwords API and forward some sort of relevant error message to the front end.

**Review by sscchan — CHANGES_REQUESTED** [review]
There is several extra .json file that looks like variants of the package.json file.
Also need to add to .gitignore of the api key module file.

File: server/lib/api_config.js (hunk 1 above)
Comment (benzeg) [reply server/lib/api_config.js]: resolved

File: server/lib/request-handler.js (hunk 6 above)
Comment (benzeg) [reply server/lib/request-handler.js:16]: resolved

File: server/lib/request-handler.js (hunk 7 above)
Comment (benzeg) [reply server/lib/request-handler.js]: resolved

File: server/lib/request-handler.js (hunk 8 above)
Comment (benzeg) [reply server/lib/request-handler.js]: resolved

File: server/lib/api_config.js (hunk 1 above)
Comment (benzeg) [reply server/lib/api_config.js]: I tried adding api_config.js and server/lib/api_config.js to gitignore but it still commits the file

**Pull request merged**